### PR TITLE
RHDEVDOCS-6207: Enable proxy awareness

### PR DESCRIPTION
**Pull request by shivanisathe25**
Version(s) for cherry-picking: build-docs-1.2

Issue:

Link to docs preview: https://84999--ocpdocs-pr.netlify.app/openshift-builds/latest/configuring/using-builds-in-a-network-restricted-environment.html

QE review:
- [ ] QE has approved this change.

SME review: @avinal 

Additional information:

### CODE REVIEW

**Review by avinal — CHANGES_REQUESTED** [review]
There should be links for OpenShift Pipelines proxy doc and Enabling cluster proxy. Is that part of a different page?

Few suggestions, rest looks good to me.

**Review by avinal — APPROVED** [review]
The content looks good to me.

File: modules/ob-verifying-cluster-wide-proxy.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// This module is included in the following assembly:
+//
+// * configuring/using-builds-in-a-restricted-environment.adoc
+
+:_mod-docs-content-type: REFERENCE
```
Comment (skrthomas) [inline modules/ob-verifying-cluster-wide-proxy.adoc:5]:
```suggestion
:_mod-docs-content-type: PROCEDURE
```

File: modules/ob-verifying-cluster-wide-proxy.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+[id="ob-verifying-cluster-wide-proxy_{context}"]
+= Verifying cluster-wide proxy
+
+Run the following command to verify if the cluster-wide proxy is correctly configured:
```
Comment (skrthomas) [inline modules/ob-verifying-cluster-wide-proxy.adoc:9]:
```suggestion
* Run the following command to verify if the cluster-wide proxy is correctly configured:
```

File: modules/ob-verifying-proxy-environment-variables.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// This module is included in the following assembly:
+//
+// * configuring/using-builds-in-a-restricted-environment.adoc
+
+:_mod-docs-content-type: REFERENCE
```
Comment (skrthomas) [inline modules/ob-verifying-proxy-environment-variables.adoc:5]:
```suggestion
:_mod-docs-content-type: PROCEDURE
```

File: modules/ob-verifying-proxy-environment-variables.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+[id="ob-verifying-the-proxy-environment-variables_{context}"]
+= Verifying the proxy environment variables
+
+Run the following command to verify the environment variables:
```
Comment (skrthomas) [inline modules/ob-verifying-proxy-environment-variables.adoc:9]:
```suggestion
* Run the following command to verify the environment variables:
```

File: modules/ob-verifying-cluster-wide-proxy.adoc (hunk 2 above)
Comment (skrthomas) [reply modules/ob-verifying-cluster-wide-proxy.adoc:9]: These two modules strike me as single-step procedures more than reference. Correct me if I am misinterpreting.

**Review by skrthomas — COMMENTED** [review]
Couple small comments. Looks great!

File: configuring/using-builds-in-a-network-restricted-environment.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+:context: using-builds-in-a-network-restricted-environment
+
+toc::[]
+
+The {pipelines-operator} provides support for injecting the proxy environment variables in the network restricted environments. As a cluster administrator, you can verify the cluster-wide proxy settings and environment variables to ensure that the proxy settings are correctly configured for {builds-title} on an OpenShift Container Platform cluster.
+
+include::modules/ob-verifying-cluster-wide-proxy.adoc[leveloffset=+1]
+include::modules/ob-verifying-proxy-environment-variables.adoc[leveloffset=+1]
+
+
```
Comment (agantony) [inline configuring/using-builds-in-a-network-restricted-environment.adoc:14]: delete extra spaces

File: configuring/using-builds-in-a-network-restricted-environment.adoc [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+[id="using-builds-in-a-network-restricted-environment"]
+include::_attributes/common-attributes.adoc[]
+= Using {builds-shortname} in a network restricted environment
+:context: using-builds-in-a-network-restricted-environment
+
+toc::[]
+
+The {pipelines-operator} provides support for injecting the proxy environment variables in the network restricted environments. As a cluster administrator, you can verify the cluster-wide proxy settings and environment variables to ensure that the proxy settings are correctly configured for {builds-title} on an OpenShift Container Platform cluster.
+
+include::modules/ob-verifying-cluster-wide-proxy.adoc[leveloffset=+1]
```
Comment (agantony) [inline configuring/using-builds-in-a-network-restricted-environment.adoc:11]: add a space b/w the include files

File: configuring/using-builds-in-a-network-restricted-environment.adoc [hunk 7]
```diff
@@ -0,0 +1,21 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="using-builds-in-a-network-restricted-environment"]
+include::_attributes/common-attributes.adoc[]
```
Comment (agantony) [inline configuring/using-builds-in-a-network-restricted-environment.adoc:3]: attributes include file should be after the title.
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata

File: modules/ob-verifying-cluster-wide-proxy.adoc (hunk 1 above)
Comment (agantony) [inline modules/ob-verifying-cluster-wide-proxy.adoc:5]: This is a procedure

File: modules/ob-verifying-cluster-wide-proxy.adoc [hunk 8]
```diff
@@ -0,0 +1,17 @@
+// This module is included in the following assembly:
+//
+// * configuring/using-builds-in-a-restricted-environment.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="ob-verifying-cluster-wide-proxy_{context}"]
+= Verifying cluster-wide proxy
+
```
Comment (agantony) [inline modules/ob-verifying-cluster-wide-proxy.adoc:8]: Add an intro sentence

File: modules/ob-verifying-cluster-wide-proxy.adoc (hunk 2 above)
Comment (agantony) [inline modules/ob-verifying-cluster-wide-proxy.adoc:9]:
```suggestion
.Procedure
* To verify if the cluster-wide proxy is correctly configured, run the following command:
```

Comment (agantony) [reply modules/ob-verifying-cluster-wide-proxy.adoc:9]: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command

File: modules/ob-verifying-proxy-environment-variables.adoc [hunk 9]
```diff
@@ -0,0 +1,26 @@
+// This module is included in the following assembly:
+//
+// * configuring/using-builds-in-a-restricted-environment.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ob-verifying-the-proxy-environment-variables_{context}"]
+= Verifying the proxy environment variables
+
+* Run the following command to verify the environment variables:
```
Comment (agantony) [inline modules/ob-verifying-proxy-environment-variables.adoc:9]:
```suggestion
.Procedure

* To verify the environment variables, run the following command:
```

Comment (agantony) [reply modules/ob-verifying-proxy-environment-variables.adoc:9]: Add an intro before the procedure

File: modules/ob-verifying-proxy-environment-variables.adoc [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+[source,terminal]
+----
+$ oc set env deployment/openshift-builds-operator --list -n openshift-builds | grep PROXY
+----
+
+The HTTP_PROXY, HTTPS_PROXY, and NO_PROXY environment variables are listed as shown in the following example:
```
Comment (agantony) [inline modules/ob-verifying-proxy-environment-variables.adoc:16]:
```suggestion
The following example output lists the `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY` environment variables:
```

Comment (agantony) [reply modules/ob-verifying-proxy-environment-variables.adoc:16]: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting:~:text=Use%20backticks%20for%20all%20non%2DGUI%20%22system%20items%22%2C%20including%3A

File: modules/ob-verifying-proxy-environment-variables.adoc [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+[source,terminal]
+----
+$ oc set env deployment/openshift-builds-operator --list -n openshift-builds | grep PROXY
+HTTP_PROXY=http://192.168.130.1:3128
+HTTPS_PROXY=https://192.168.130.1:3129
+NO_PROXY=.cluster.local,.svc,.testing,10.217.0.0/22,10.217.4.0/23,127.0.0.1,192.168.126.0/24,192.168.1
+30.11,api-int.crc.testing,localhost
+----
```
Comment (agantony) [inline modules/ob-verifying-proxy-environment-variables.adoc:25]:
```suggestion
.Example output
[source,terminal]
----
HTTP_PROXY=http://192.168.130.1:3128
HTTPS_PROXY=https://192.168.130.1:3129
NO_PROXY=.cluster.local,.svc,.testing,10.217.0.0/22,10.217.4.0/23,127.0.0.1,192.168.126.0/24,192.168.1
30.11,api-int.crc.testing,localhost
----
```

**Review by agantony — COMMENTED** [review]
Added a few nits; I need someone to take a second look at my comments. Please do not add the merge review needed label back again.

You will also need to fix the alignment issues in the preview.

File: modules/ob-verifying-cluster-wide-proxy.adoc (hunk 8 above)
Comment (shivanisathe25) [reply modules/ob-verifying-cluster-wide-proxy.adoc:8]: Introduction to both the subsections (what a cluster admin can do) is included in the main introduction itself

**Comment by shivanisathe25** [comment]
> I do not see a QE approval?

the team doesnt have a QE as of now. Hence, we are just considering SME approval.

**Comment by agantony** [comment]
Also, please squash all your commits.

**Comment by agantony** [comment]
Add continous "+" sign after the code blocks:
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output:~:text=If%20you%20must%20add%20any%20text%2C%20admonitions%2C%20or%20code%20blocks%20you%20have%20to%20add%20the%20continuous%20%2B%2C%20as%20shown%20in%20the%20example%3A 

This will fix the alignment in the preview.

**Review by agantony — COMMENTED** [review]
A few more nits.

File: modules/ob-verifying-cluster-wide-proxy.adoc [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+:_mod-docs-content-type: PROCEDURE
+[id="ob-verifying-cluster-wide-proxy_{context}"]
+= Verifying cluster-wide proxy
+
+As a cluster administrator, you can verify the cluster-wide proxy settings.
+
+.Procedure
+
+* To verify if the cluster-wide proxy is correctly configured, run the following command:
+
```
Comment (agantony) [inline modules/ob-verifying-cluster-wide-proxy.adoc:14]:
```suggestion
+
```

File: modules/ob-verifying-proxy-environment-variables.adoc [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+:_mod-docs-content-type: PROCEDURE
+[id="ob-verifying-the-proxy-environment-variables_{context}"]
+= Verifying the proxy environment variables
+
+As a cluster administrator, you can verify the environment variables to ensure that the proxy settings are correctly configured for {builds-title} on an OpenShift Container Platform cluster
+
+.Procedure
+
+* To verify the environment variables, run the following command:
+
```
Comment (agantony) [inline modules/ob-verifying-proxy-environment-variables.adoc:14]:
```suggestion
+
```

**Review by agantony — COMMENTED** [review]
2 more comments that will fix the alignment.

File: modules/ob-verifying-proxy-environment-variables.adoc [hunk 14]
```diff
@@ -0,0 +1,30 @@
+// This module is included in the following assembly:
+//
+// * configuring/using-builds-in-a-restricted-environment.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ob-verifying-the-proxy-environment-variables_{context}"]
+= Verifying the proxy environment variables
+
+As a cluster administrator, you can verify the environment variables to ensure that the proxy settings are correctly configured for {builds-title} on an OpenShift Container Platform cluster
```
Comment (agantony) [inline modules/ob-verifying-proxy-environment-variables.adoc:9]:
```suggestion
As a cluster administrator, you can verify the environment variables to ensure that the proxy settings are correctly configured for {builds-title} on an OpenShift Container Platform cluster.
```

**Review by agantony — COMMENTED** [review]
One small nit

File: modules/ob-verifying-proxy-environment-variables.adoc (hunk 14 above)
Comment (shivanisathe25) [reply modules/ob-verifying-proxy-environment-variables.adoc]: done!

**Comment by agantony** [comment]
The new PR has overtaken the changes.
https://github.com/openshift/openshift-docs/pull/85319

**Pull request closed without merging**